src/Formatter/FormatterManager.php

```diff
@@ -23,9 +24,18 @@ class FormatterManager implements FormatterManagerInterface {
   protected $plugins;
 
   /**
+   * The resource.
+   *
+   * @todo: Remove this coupling.
```

Comment:
I feel that the resource plugin should deliver the renderable data to the formatter. Then the formatter only has to create a return string in the right format.

If the formatter needs further information for complex formatting, then we should provide a context object of some sort, with that information, without coupling both objects. My argument for that is that the resource is much more than the context for formatting output, therefore it seems an unnecessary coupling between the objects and an overkill.

Anyways, as you see it's just a TODO. And not the most important one.